core/engine_test.go

```diff
@@ -141,7 +146,7 @@ func TestEngineRun(t *testing.T) {
 		defer test.wait()
 
 		startTime := time.Now()
-		assert.ErrorContains(t, test.run(), "test run aborted by signal")
+		assert.ErrorContains(t, test.run(), "context deadline exceeded")
```

Comment:
I realize this test is commented out in #2815, but why does this error message change? If execution is not aborted by a signal anymore and we time out instead, that's a functional change.

Reply:
Because before this test run, the `Engine` was making an assumption that, when its `runCtx` context was cancelled, that meant an external signal aborted it. See this code from `master`: https://github.com/grafana/k6/blob/daa866aca990a3424f9e97af8f9177b17347fe61/core/engine.go#L191-L196

And this is what the test checks, in a very clumsy way, with the 100ms duration above. Now, since we don't have any of that special handling in the Engine, cancelling the context in such a simple way will just return a context cancelled error. I can probably fix the test, but I'd need to rewrite it, and it's very specific for the `Engine` and goes away in the next PR anyway. Instead, I have tests in [cmd_run_test.go](https://github.com/grafana/k6/pull/2813/files#diff-01466ed7a6fc9af57eb5a2b47cccb5b6a538e5de9eb3fcd2c3c722e4abe233a4) that check the same thing.